src/distribution/logistic.rs

```diff
@@ -0,0 +1,454 @@
+    ///
+    /// ```
+    /// use statrs::distribution::Logistic;
+    ///
+    /// let mut result = Logistic::new(0.0, 1.0);
+    /// assert!(result.is_ok());
+    ///
+    /// result = Logistic::new(0.0, -1.0);
+    /// assert!(result.is_err());
+    /// ```
+    pub fn new(location: f64, scale: f64) -> Result<Logistic> {
+        if location.is_infinite()
+            || location.is_nan()
+            || scale.is_infinite()
+            || scale.is_nan()
+            || scale <= 0.0
+        {
+            Err(StatsError::BadParams)
+        } else {
+            Ok(Logistic { location, scale })
+        }
+    }
+
+    /// Returns the location of the logistic distribution
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use statrs::distribution::Logistic;
+    ///
+    /// let n = Logistic::new(0.0, 1.0).unwrap();
+    /// assert_eq!(n.location(), 0.0);
+    /// ```
+    pub fn location(&self) -> f64 {
+        self.location
+    }
+
+    /// Returns the scale of the logistic distribution
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use statrs::distribution::Logistic;
+    ///
+    /// let n = Logistic::new(0.0, 1.0).unwrap();
+    /// assert_eq!(n.scale(), 1.0);
+    /// ```
+    pub fn scale(&self) -> f64 {
+        self.scale
+    }
+}
+
+impl ::rand::distributions::Distribution<f64> for Logistic {
+    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> f64 {
+        let x: f64 = rng.gen_range(0.0..1.0);
+        self.location + self.scale * logistic::logit(x)
+    }
+}
+
+impl ContinuousCDF<f64, f64> for Logistic {
+    /// Calculates the cumulative distribution function for the
+    /// logistic distribution at `x`
+    ///
+    /// # Formula
+    ///
+    /// ```ignore
+    /// 1 / (1 + exp(-(x - μ) / σ))
+    /// ```
+    ///
+    /// where `μ` is the location, `σ` is the scale
+    fn cdf(&self, x: f64) -> f64 {
+        logistic::logistic((x - self.location()) / self.scale)
+    }
+    /// Calculates the inverse cumulative distribution function for the
+    /// logistic distribution at `p`
+    ///
+    /// # Formula
+    ///
+    /// ```ignore
+    /// μ + σ * ln(p / (1 - p))
+    /// ```
+    ///
+    /// where `μ` is the location, `σ` is the scale
+    fn inverse_cdf(&self, p: f64) -> f64 {
+        self.location + self.scale * logistic::logit(p)
+    }
+}
+
+impl Min<f64> for Logistic {
+    /// Returns the minimum value in the domain of the logistic
+    /// distribution representable by a double precision float
+    ///
+    /// # Formula
+    ///
+    /// ```ignore
+    /// NEG_INF
+    /// ```
+    fn min(&self) -> f64 {
+        f64::NEG_INFINITY
+    }
+}
+
+impl Max<f64> for Logistic {
+    /// Returns the maximum value in the domain of the logistic
+    /// distribution representable by a double precision float
+    ///
+    /// # Formula
+    ///
+    /// ```ignore
+    /// INF
+    /// ```
+    fn max(&self) -> f64 {
+        f64::INFINITY
+    }
+}
+
+impl Distribution<f64> for Logistic {
```

Comment:
It seems we're using dedicated traits for some things like min and max, whereas others are nested inside this more general Distribution trait. Could it make sense to only do one or the other?

Reply:
The reason these have not been coalesced is because it was not clear to me how useful these other properties outside of `Distribution` were at the time I wrote the traits into their current form. Max and min for example have no generalization to higher dimensions. `Mode` might be a (n-1)D subset if distribution is nD.
I can't think of a reason why I didn't include `Median` into `Distribution` though, so I could revisit that. I'm generally speaking for fewer traits, because I don't see traits here pulling their weight. How to properly integrate higher dimensional distributions is also unfinished.

Reply:
That makes sense, thanks for the explanation @troublescooter